src/colorizer/FrameCache.ts

```diff
@@ -69,78 +75,66 @@ export default class FrameCache {
     this.setFirst(entry);
   }
 
-  public get length(): number {
-    return this.data.length;
+  public get size(): number {
+    return this.currentSize;
   }
 
-  /** Inserts a frame into the cache at `index` */
-  public insert(index: number, frame: Texture): void {
-    if (index >= this.data.length) {
-      return;
+  /**
+   * Inserts a value into the cache with the key. If the key already exists, the value is replaced.
+   * If the cache is full, the least recently used entry is evicted and its `dispose()` function is
+   * called if it exists.
+   *
+   * Throws an error if the `size` is greater than the cache's `maxSize`, as defined in the
+   * constructor.
+   */
+  public insert(key: string | number, value: E, size: number = 1): void {
```

Comment:
it could also be interesting if the `E` type had an optional function it could call to return its size, rather than explicitly passing it in separately.  (Both scenarios are potentially interesting. I'm thinking about things like Arrays or ArrayBuffers that know their size)

Reply:
Ended up skipping this for now, because I think it's more general if the size is an optional property on the insertion method than on the objects themselves. This is definitely making the assumption that entries do not change size.